File: Classification Program/memahami_push_button.m
function memahami_push_button()

myPi=raspi('169.254.29.168','pi','raspberry');
LED_PIN=21;
PUSH_BUTTON=20;

configurePin(myPi,LED_PIN,'DigitalOutput');
configurePin(myPi,PUSH_BUTTON,'DigitalInput');

while true
    if readDigitalPin(myPi,PUSH_BUTTON)==1
        writeDigitalPin(myPi,LED_PIN,1);
    else
        writeDigitalPin(myPi,LED_PIN,0)
    end
end

end

File: Classification Program/codegen/exe/memahami_push_button/memahami_push_button.cpp
//
//  Student License - for use by students to meet course requirements and
//  perform academic research at degree granting institutions only.  Not
//  for government, commercial, or other organizational use.
//
//  memahami_push_button.cpp
//
//  Code generation for function 'memahami_push_button'
//


// Include files
#include "memahami_push_button.h"
#include "SDLVideoDisplay.h"
#include "memahami_push_button_data.h"
#include "memahami_push_button_initialize.h"
#include "raspi.h"

// Function Definitions
void memahami_push_button()
{
  coder::b_raspi myPi;
  if (!isInitialized_memahami_push_button) {
    memahami_push_button_initialize();
  }

  myPi.RaspiDisplay.matlabCodegenIsDeleted = true;
  myPi.matlabCodegenIsDeleted = true;
  myPi.init();
  myPi.configurePin();
  myPi.b_configurePin();
  while (1) {
    if (myPi.readDigitalPin()) {
      myPi.writeDigitalPin();
    } else {
      myPi.b_writeDigitalPin();
    }
  }
}

// End of code generation (memahami_push_button.cpp)
